modin/experimental/engines/pyarrow_on_ray/frame/partition.py

```diff
@@ -25,7 +25,18 @@ def to_pandas(self):
         Returns:
             A Pandas DataFrame.
         """
-        dataframe = self.get().to_pandas()
+        import os
+
+        if os.name == "posix":
+            # workaround for segmentation fault
+            # for details see #2347
+            from pyarrow import Table
+
+            source_df = self.get()
+            pydict = source_df.to_pydict()
+            dataframe = Table.from_pydict(pydict, schema=source_df.schema).to_pandas()
```

Comment:
Could you please explain the logic? I don't like workarounds, especially those which hide issues which I don't understand a root cause for.

Reply:
> Could you please explain the logic? I don't like workarounds, especially those which hide issues which I don't understand a root cause for.

This was done to make sure we don't have other problems. As for given changes - for some reasons there is a segfault only on Linux for specific data conversion during arrow_to_pandas conversion. The reason is not clear at the moment

Reply:
According to disassembly you posted below I would say we're just getting lucky on Windows that nothing crashes. I would prefer us to investigate the issue instead of sweeping it under the rug by using this workaround (which is wasting quite a few CPU time along the way).

Reply:
> According to disassembly you posted below I would say we're just getting lucky on Windows that nothing crashes. I would prefer us to investigate the issue instead of sweeping it under the rug by using this workaround (which is wasting quite a few CPU time along the way).

In progress...